assets/in

```diff
@@ -99,8 +100,11 @@ if [ ! "${encryption_key}" = "" ];then
   echo "...done."
 fi
 
-version=$(echo "${payload}" | jq -re '.version')
+file_info=$(aws --endpoint-url ${endpoint_url} ${ssl_verification_opt} ${timeouts} --output json s3api list-objects --bucket ${bucket} --prefix ${path} | jq '[.Contents[] | {"file_name": .Key, "last_changed": .LastModified, "file_size": .Size}]')
 
-jq -n "{
-  version: ${version}
-}" >&3
+# shellcheck disable=SC2236
+if [ ! -z "${file_info}" ]; then
```

Comment:
! -z == -n :)

Reply:
Please correct me if I'm wrong but this  condition is a  bit odd in my opinion.

if file_info is not empty then it executes:
`echo "${payload}" | /opt/resource/metadata.py "some_string" >&3`
if it's empty it executes:
`echo "${payload}" | /opt/resource/metadata.py (nothing_here) >&3`

Let's consider how this will behave without this condition by just executing this line:
```
echo "${payload}" | /opt/resource/metadata.py "${file_info}" >&3
```
if file_info is not empty then it executes:
`echo "${payload}" | /opt/resource/metadata.py "some_string" >&3`
if it's empty it executes:
`echo "${payload}" | /opt/resource/metadata.py (nothing_here) >&3`

Isn't it the same?

Reply:
the -n wasn't being interpreted the same as ! -z when we tested, i did try that initially. The if triggered when it shouldn't have, whereas when we tried with ! -z it triggered when we expected. I can test again and show you what i mean. 

Regarding the other comment, that may be the case, but it works with the checks we implemented in the python script:
if len(sys.argv) > 1:
            file_info=json.loads(sys.argv[1])

Potentially we could change it in the assets/in script but we would have to change the python script logic too. Not sure which is best..